# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [6]:
#1
a=[]
for i in range(5):
    n=i+1
    a.append(3*n**2+7)
    print(f'a_{n}={a[i]}')

a_1=10
a_2=19
a_3=34
a_4=55
a_5=82


In [7]:
#2
a=[]
for i in range(5):
    n=i+16
    a.append(3*n**2+7)
    print(f'a_{n}={a[i]}')

a_16=775
a_17=874
a_18=979
a_19=1090
a_20=1207


# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [20]:
#1

a=[]

for i in range(4):
    n=i+1
    a.append(5*n**3)
    print(f'a_{n}={a[i]}')


a_1=5
a_2=40
a_3=135
a_4=320


In [24]:
#2

a = []

for i in range(4):
    n=i+9
    a.append(5*n**3)
    print(f'a_{n}={a[i]}')


a_9=3645
a_10=5000
a_11=6655
a_12=8640


In [27]:
#3

final = 40000
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = 5*n**3
    dif = final-termino
print(f'El termino {final} esta en la posicion {n}')

El termino 40000 esta en la posicion 20


# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

In [40]:
#1
a=[]
for i in range(1):
    n=i+10
    a.append(2*n)
    print(f'a_{n}={a[i]}')

a_10=20


In [42]:
#2
par = []

for i in range(100):
    n=i+1
    par.append(2*n)
suma = sum(par)

print(f'La suma de 100 numeros pares es {suma}')

La suma de 100 numeros pares es 10100


In [ ]:
#3
posicion = par.index(58)
print(f'3) El termino 58 se encuentra en la posicion {posicion + 1} de la sucesion')


3) El termino 58 se encuentra en la posicion 29 de la sucesion


# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

In [17]:
#1

g=[]
for i in range(6):
    n=i+1
    g.append(500 * 1.15**(n-1))
print(f'a_{n}={g[i]:.0f}')

a_6=1006


Al cabo de 3 meses tendra 661 usuarios aprox y a los 6 meses 1006 aprox

In [18]:
#2

g=[]
for i in range(12):
    n=i+1
    g.append(500 * 1.15**(n-1))
print(f'a_{n}={g[i]:.0f}')

a_12=2326


In [17]:
#3

g=[]
for i in range(12):
    n=i+1
    g.append(500 * 1.15**(n-1))
print(f'a_{n}={g[i]:.0f}')

suma = sum(g)

print(f'Luego de un año, el software tiene {suma:.0f} usuarios')

a_12=2326
Luego de un año, el software tiene 14501 usuarios


# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita $\$12.000$ y cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito mensual del estudiante. Determine:

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

In [16]:
#1

a=[]
for i in range(12):
    n=i+1
    a.append(12000 + (n-1)* 2000)
print(f'a_{n}={a[i]:.0f}')

a_12=34000


al primer año el deposito total es de 34000

2) AL ahorrar por un 1 año y 2 meses (febrero del siguiente año) es de 38000

In [12]:
#3

a=[]
for i in range(24):
    n=i+1
    a.append(12000 + (n-1)* 2000)
suma=sum(a)

print(f'Al finalizar el ahorro de 2 años el tendra un total de {suma}')

Al finalizar el ahorro de 2 años el tendra un total de 840000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

In [15]:
#1

g=[]
for i in range(5):
    n=i+1
    g.append(3 * 0.9**(n-1))
print(f'a_{n}={g[i]:.1f}')

a_5=2.0


In [9]:
#2

g=[]
for i in range(10):
    n=i+1
    g.append(3 * 0.9**(n-1))
suma=sum(g)

print(f'la duracion total del proyecto despues de las 10 fases es de {suma:.0f}')

la duracion total del proyecto despues de las 10 fases es de 20


# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

1
El primer termino es -50

2
an = -56 + (n+1) * 4

In [28]:
#3

a=[]
for i in range(1000):
    n=i+1
    a.append(-56 + (n-1)* 4)
print(f'El termino que ocupa el lugar 100 es {a[99]:.1f}')

#4

posicion = a.index(1680)
print(f' El termino 1680 se encuentra en la posicion {posicion + 1} de la sucesion')

El termino que ocupa el lugar 100 es 340.0
El termino 1680 se encuentra en la posicion 435 de la sucesion


# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
4. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

1.
a3 = 5

a6 = 40

a3 = a1 * r(3-1)
5 = a1 * r**2

a6 = a1 * r(6-1)
40 = a1*r**5

2 = r

2.
5 = a1 * 2**2
5/4 = a1

3.
an = 1.25 * 2**(n-1)

In [38]:
#4

g=[]
for i in range(1000):
    n=i+1
    g.append(1.25*2**(n-1))
print(f'El termino que ocupa el lugar 8 es {g[7]:.0f}')

posicion = g.index(20480)
print(f'El termino 20480 se encuentra en la posicion {posicion + 1} de la sucesion')


El termino que ocupa el lugar 8 es 160
El termino 20480 se encuentra en la posicion 15 de la sucesion


# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello, mide cada mes las transacciones por segundo (TPS) de sus servidores. Se ha obtenido que el segundo mes, el rendimiento del servidor es $1,2$ miles de TPS y el quinto mes, el rendimiento es de $2,7$ miles de TPS. Determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

a2 = 1,2

a5 = 2,7

1) 1,2 =  a1 + d

2) 2,7 = a1 + 4d

Restamos 2) - 1) las ecuaciones , quedando

1.5 = 3d
0.5 = d

Remplazo d = 0.5 en la ecuacion 1)

1.2 = a1 + o,5 despejamos a1
1.2 -0.5 = a+1
0.7 = a1

an = 0.7 + (n-1)*0.5

In [16]:
#1
a = []

for i in range(100):
    n = i+1
    a.append(0.7 +(n-1)*0.5)
print(f'El rendimiento del noveno mes es {a[8]} miles de TPS')

El rendimiento del noveno mes es 4.7 miles de TPS


In [19]:
#2

posicion = a.index(23.4)
print(f'Cuando el rendimiento es de 23.4 miles de TPS, en el {posicion +1} mes')

ValueError: 23.4 is not in list

# Problema 11

Una aplicación de redes sociales inició el $1$ de enero y comenzó a ganar usuarios rápidamente. El $3$ de enero tenía $500$ usuarios y el $10$ de enero tenía $1.501$ usuarios. Si el número de usuarios sigue creciendo de manera constante, determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



a3 = 500
a10 = 1501

500 = a1 + (3-1)*d
1501 = a1 + (10-1)*d

1) 500 = a1 + (3-1)*2
2) 1501 = a1 + (10-1)*9

Restar 2) con 1)

1001 = 7d

1001/7 = d
143 = d

500      = a1 + 2*143
500 -286 = a1
    214 = a1

an = 214 + (n-1)*143

In [24]:
#2
a = []

for i in range(40):
    n = i+1
    a.append(214 + (n-1)*143)
print(f'Los usuarios que tendra la aplicacion son {a[30]} en enero')


Los usuarios que tendra la aplicacion son 4504 en enero


In [26]:
posicion = a.index(5505)
print(f'Los meses que lleva funcionando el sistema cuando son 5505 usuarios son {posicion + 1} ')

Los meses que lleva funcionando el sistema cuando son 5505 usuarios son 38 


# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

a2 = 402000
a3 = 404010

a3/a2 = r
1,005= r

a2/a1 = 1,005

402000/1,005 = a1
400.000 = a1

funcion: an = 400.000 * 1.005**(n-1)

In [40]:
#1 y #2
g=[]
for i in range(25,37):
    n=i+1
    g.append(400000 * 1.005**(n-1))

suma = sum(g)

print(f'El ingreso total de Javier al tercer año de trabajo es {suma:.0f}')

print(f'El salario aproximado que recibira javier es ${g[9]:.0f}')


El ingreso total de Javier al tercer año de trabajo es 5589468
El salario aproximado que recibira javier es $473921


# Problema 13

a1 = 30

a2 = 60

a3= 120

a3/a2 = 2
a2/a1 = 2

r = 2

a1 = 30

In [47]:
#1

g=[]
for i in range(13):
    n=i+1
    g.append(30 * 2**(n-1))

suma = sum(g)

print(f'La capacitacion total  desde el conmienzo al final  de los 3 años es de {suma:.0f}')
print(f'Al octavo trimestre la cantidad de trabajadores es de {g[7]:.0f}')




La capacitacion total  desde el conmienzo al final  de los 3 años es de 245730
Al octavo trimestre la cantidad de trabajadores es de 3840


Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.

a4 = 128
a9 = 88

128 = a1 + (4-1)*d
88 = a1 + (9-1)*d

128= a1 + 3d
88 = a1 8d



-40 = 5d

-40/5 = -8

-8= d

128 = a1 + 3*-8
128+ 24 = a1
152= a1

an = 152 + (n-1)* -8


In [53]:
#1 y #2

a = []

for i in range(20):
    n = i+1
    a.append(152 + (n-1)* -8)

posicion = a.index(0)
print(f'a los {posicion + 1} minutos deja de recibir agua del tambor')

suma = sum(a)
print(f' La cantidad total que recibe el tambor es de {suma}')


a los 20 minutos deja de recibir agua del tambor
 La cantidad total que recibe el tambor es de 1520
